components/molecules/HighlightInput/highlight-input-form.tsx

```diff
@@ -500,6 +500,13 @@ const HighlightInputForm = ({ refreshCallback }: HighlightInputFormProps): JSX.E
   }, 250);
 
   useEffect(() => {
+    const queryParams = new URLSearchParams(window.location.search);
```

Comment:
Refactor (blocking): Instead of `new URLSearchParams(window.location.search)` prefer the `useRouter` hook.

e.g.

```
  const router = useRouter();
  const { prurl } = router.query;
```

Also, this doesn't need to be in the a `useEffect`.

Reply:
Thanks for the suggestion, I would love to get help with this. 

- Out of curiosity: why would you suggest the useRouter? 

When I am using the useRouter outside the useEffect it gives me too many rendering error; how could I solve that? 

```
Error: Too many re-renders. React limits the number of renders to prevent an infinite loop.
```
here is the code: 
```js
const router = useRouter();
const { prurl } = router.query;
if (prurl) {
  setHighlightLink(prurl as string);
}
```

Reply:
I could get it to work by using this:

```js
  const router = useRouter();
  const { prurl } = router.query;
  useEffect(() => {
    if(!prurl) return;
    setHighlightLink(prurl as string);
  }, [prurl]);
```

Reply:
Yeah, that's my bad. I think it needs to remain in a `useEffect` like you have. I suggested the `useRouter` as we're using pretty much everywhere for stuff like this.

Reply:
I have worked on this for almost an hour yesterday lol. but it was worth it, my previous logic were broken to be honest. and I think the userRouter is more suited right now.